shell/platform/darwin/ios/framework/Source/platform_views_controller.h

```diff
@@ -0,0 +1,402 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef FLUTTER_SHELL_PLATFORM_DARWIN_IOS_FRAMEWORK_SOURCE_PLATFORM_VIEWS_CONTROLLER_H_
+#define FLUTTER_SHELL_PLATFORM_DARWIN_IOS_FRAMEWORK_SOURCE_PLATFORM_VIEWS_CONTROLLER_H_
+
+#include <Metal/Metal.h>
+
+#include "flutter/flow/surface.h"
+#include "flutter/fml/memory/weak_ptr.h"
+#include "flutter/fml/platform/darwin/scoped_nsobject.h"
+#include "flutter/fml/trace_event.h"
+#import "flutter/shell/platform/darwin/common/framework/Headers/FlutterChannels.h"
+#import "flutter/shell/platform/darwin/ios/framework/Headers/FlutterPlatformViews.h"
+#import "flutter/shell/platform/darwin/ios/framework/Headers/FlutterPlugin.h"
+#import "flutter/shell/platform/darwin/ios/framework/Source/FlutterViewResponder.h"
+#import "flutter/shell/platform/darwin/ios/ios_context.h"
+#include "fml/task_runner.h"
+#include "impeller/base/thread_safety.h"
+#include "third_party/skia/include/core/SkRect.h"
+
+@class FlutterTouchInterceptingView;
+@class FlutterClippingMaskViewPool;
+
+namespace flutter {
+
+class IOSSurface;
+
+/// @brief State holder for a Flutter overlay layer.
+struct OverlayLayer {
```

Comment:
Move this and the pool to its own TU, OOL the cdtors, and add a headerdoc while you are at it too?

Reply:
Done